docs: clarify USER command explanation (#2956)
moby/buildkit: frontend/dockerfile/docs/reference.md

```

The `USER` instruction sets the user name (or UID) and optionally the user
group (or GID) to use as the default user and group for the remainder of the

> current stage. This user will be used in `RUN` instructions, and will be the
> owner of files and directories created as part of `ADD`/`COPY`. Additionally, at
> runtime, this user will run the relevant `ENTRYPOINT` and `CMD` commands.

Reviewer: ```suggestion
current stage. The specified user is used on `RUN` instructions, is the owner of 
any files or directories created as part of `ADD`/`COPY`. Additionally, at
runtime, this user runs the relevant `ENTRYPOINT` and `CMD` commands.
```
